HealthcareSystem/IntegrationAdaptersTests/UnitTests/ApiRegisterTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Backend.Repository;
+using IntegrationAdapters.Controllers;
+using IntegrationAdaptersTests.DataFactory;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Xunit;
+using Moq;
+
+namespace IntegrationAdaptersTests.UnitTests
+{
+    public class ApiRegisterTests
+    {
+        [Fact]
+        public void Successfully_registers_api()
+        {
+            var pharmacy = CreatePharmacy.CreateValidTestObject();
+            var stubRepository = new Mock<IPharmacyRepo>();
+            stubRepository.Setup(m => m.CreatePharmacy(pharmacy));
+            var tempData = new TempDataDictionary(new DefaultHttpContext(), Mock.Of<ITempDataProvider>())
+            {
+                ["Success"] = "Registration successful!"
+            };
+            var controller = new HomeController(stubRepository.Object)
+            {
+                TempData = tempData
+            };
+
+            var result = controller.ApiRegister(pharmacy);
+
+            Assert.IsType<RedirectToActionResult>(result);
+        }
+
+        [Fact]
+        public void Does_not_successfully_register_api()
+        {
+            var pharmacy = CreatePharmacy.CreateInvalidTestObject();
+            var stubRepository = new Mock<IPharmacyRepo>();
+            stubRepository.Setup(m => m.CreatePharmacy(pharmacy));
+            var tempData = new TempDataDictionary(new DefaultHttpContext(), Mock.Of<ITempDataProvider>())
+            {
+                ["Success"] = "Registration successful!"
+            };
+            var controller = new HomeController(stubRepository.Object)
+            {
+                TempData = tempData
+            };
+            controller.ModelState.AddModelError("Url", "Required");
+
+            var result = controller.ApiRegister(pharmacy);
+
+            Assert.IsType<BadRequestObjectResult>(result);
```

Comment:
What is being tested in these exactly?  I find it kind of hard to tell from the code.

Reply:
It's testing if the method in the controller checks correctly if the model passed is valid. That's why I just assert depending on what kind of request it returns. Also TempData stuff is used for the success message in frontend and I have to initialize it here else the object for it never gets created and the test fails.

Reply:
So it essentially tests if your model is annotated correctly, right? Since MVC does the rest of it automatically if it is? If so I suppose that's alright, but I don't really like how much it's tied into MVC (I'm not the biggest fan of how easily the MVC model annotations in the backend can get mixed up with the Entity Framework annotations either, but I suppose there's no easy way to fix that right now).

Reply:
Yeah basically checks are there in case someone manages to bypass frontend validation and this was the best solution with what we've got right now, could be refactored at a later point if needed